File: BACKEND/Route-Service/Services/Stop/StopService.cs
using Gridify;
using Mapster;
using Route_Service.Reposetories.Stop;
using Shared.Dtos;

namespace Route_Service.Services.Stop
{
    public class StopService : IStopService
    {
        private readonly IStopRepo _stopRepo;
        private readonly ILogger<StopService> _logger;

        public StopService(IStopRepo stopRepo ,ILogger<StopService> logger)
        {
            _stopRepo = stopRepo;
            _logger = logger;
        }

        public async Task<StopResponse> AddStop(StopRequest stopRequest)
        {
            var stop = await _stopRepo.AddStop(stopRequest);
            return stop;
        }

        public Paging<StopResponse> GetStops(GridifyQuery gridifyQuery)
        {
            var stops = _stopRepo.GetAllStops().Gridify(gridifyQuery);
            return stops.Adapt<Paging<StopResponse>>(); 
        }

        public async Task<StopResponse> GetStopById(int id)
        {
            var stop = await _stopRepo.GetStopById(id);
            return stop.Adapt<StopResponse>();  
        }

        public async Task<StopResponse> UpdateStop(int id,StopRequest stopRequest)
        {
            var stop = await _stopRepo.UpdateStop(id, stopRequest);
            return stop;
        }

        public async Task DeleteStop(int id)
        {
            await _stopRepo.DeleteStop(id);
        }
    }
}

File: FRONTEND/src/app/features/Stops/services/stop.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { GridifyRequest, GridifyResponse } from '../../../shared/types/Dtos/gridify.dto';
import { StopRequest, StopResponse } from '../../../shared/types/Dtos/stop.dto';
import { BehaviorSubject, Observable, Subject } from 'rxjs';
import {  makeRequestParams } from '../../../shared/helpers/GridifyRequestBuilder';
import { env } from '../../../../environments/environment.dev';
import { GridifyBuilder } from '../../../shared/helpers/GridifyBuilder';

@Injectable({
  providedIn: 'root'
})
export class StopService {
  

  constructor(private client:HttpClient) { }
    // url:string = `${env.apiBaseUrl}/routes/stop`;
    url:string = `https://localhost:7025/api/route/Stop`;


  getStops(query:GridifyBuilder):Observable<GridifyResponse<StopResponse>>{
    const params = makeRequestParams(query);
    return this.client.get<GridifyResponse<StopResponse>>(this.url,{params}); 
  }

  addStop(stop:StopRequest):Observable<StopResponse>{
    return this.client.post<StopResponse>(this.url,stop);
  }

  getStopById(id:number):Observable<StopResponse>{
    return this.client.get<StopResponse>(`${this.url}/${id}`);
  }

  deleteStop(id:number){
    return this.client.delete(`${this.url}/${id}`);
  }

  updateStop(stopReq:StopRequest,id:number){
    return this.client.put(`${this.url}/${id}`,stopReq);
  }

}
